### ISSUE TITLE
Intelligent Feature Engineering based on column name

### ISSUE THREAD

**Issue by ParthivNaresh** [alteryx/evalml#2010]
Considering that many use cases exist which leverage data that is commonly integrated throughout their respective domains, such as debt and income levels for financial services, or phase trials and patients recruited for biotech, it would help to have the ability to automatically identify and engineer features based on their data type and/or name.

For example:
-  A "Name" category identified in a dataset would automatically be put through a process whereby "Mr.", "Miss", "Mrs." etc would help generate new features based on gender.
- An "Income" (as opposed to "Expenditures") category would expect values to be 0 or above, and would automatically drop values below 0 or would cap them at 0.
- An "Education" category would provide Ordinal Encoding to assign numerical values to expected educational levels, such as 0 for drop out, 1 for associates degree, 2 for bachelors degree, etc.

This would need a design document to determine the scope and default use cases to initially build for. A follow up issue could be filed for integrating it into AutoML to provide automated feature engineering for datasets that have matching column names to the ones we're looking for.

**Comment by dsherry** [comment]
Yep! In icebox for now. This feels like it should be a woodwork feature (@gsheni )
